dbt_style_guide.md

```diff
@@ -6,87 +6,148 @@ Our models (typically) fit into three main categories: staging, marts, base/inte
+
+- For `base` or `staging` models, columns should be ordered in categories, where identifiers are first and date/time fields are at the end.  
+  Example:
+  ```sql
+  transformed as (
+
+      select
+
+        -- ids
+        order_id,
+        customer_id,
+
+        -- dimensions
+        order_status,
+
+        -- numeric
+        order_total,
+
+        -- boolean
+        is_shipped,
+
+        -- date/times
+        created_at,
+        updated_at,
+        _sdc_batched_at
+
+      from source
+
+  )
+  ```
+
+- Date/time columns should be named according to these conventions:
+  - Timestamps: `<event>_at`
+    Format: UTC  
+    Example: `created_at`
+  
+  - Dates: `<event>_date`  
+    Format: Date  
+    Example: `created_date`
+
+- Booleans should be prefixed with `is_` or `has_`.
+
+- Price/revenue fields should be in decimal currency (e.g. `19.99` for $19.99; many app databases store prices as integers in cents). If non-decimal currency is used, indicate this with suffix, e.g. `price_in_cents`.
+
+- Avoid reserved words as column names.
+
+- Consistency is key! Use the same field names across models where possible.  
+Example: a key to the `customers` table should be named `customer_id` rather than `user_id`.
 
 ## CTEs
 
-For more information about why we use so many CTEs, check out [this discourse post](https://discourse.getdbt.com/t/why-the-fishtown-sql-style-guide-uses-so-many-ctes/1091).
+For more information about why we use so many CTEs, check out [this glossary entry](https://docs.getdbt.com/terms/cte).
 
-- All `{{ ref('...') }}` statements should be placed in CTEs at the top of the file
 - Where performance permits, CTEs should perform a single, logical unit of work.
-- CTE names should be as verbose as needed to convey what they do
-- CTEs with confusing or noteable logic should be commented
-- CTEs that are duplicated across models should be pulled out into their own models
-- create a `final` or similar CTE that you select from as your last line of code. This makes it easier to debug code within a model (without having to comment out code!)
-- CTEs should be formatted like this:
+
+- CTE names should be as verbose as needed to convey what they do.
+
+- CTEs with confusing or noteable logic should be commented with SQL comments, as you would with any complex functions.
+
+- CTEs that are duplicated across models should be pulled out and created as their own models.
+
+- CTEs and SQL within a model should follow this structure:
+  - `with` statement
+  - Import CTEs
+  - Logical CTEs
+  - Final CTE
+  - Simple `select` statement
+
+- All `{{ ref('...') }}` statements should be placed in CTEs at the top of the file ("import" CTEs)
+
+- SQL should end with a simple select statement. All other logic should be contained within CTEs.  
+  Example: `select * from final`
+
+- Where applicable, opt for filtering within import CTEs over filtering within logical CTEs.

```

Comment:
Continuing on this thought, there's some research that was conducted last year by Dominik Golebiewski that suggests that in some circumstances it can be considerably more performant to have duplicated import CTEs with separate filters than to have separate logical CTEs that filter from a commont import CTE.

https://medium.com/@AtheonAnalytics/snowflake-query-optimiser-unoptimised-cf0223bdd136

Reply:
This is where I think we should leave this as our "default style" that we're happy to implement unless we run in to scenarios like the two above. Knowing the things both of you have said is part of our expertise, so I don't feel that we need to elaborate on this because there are the words "where applicable"!